processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -149,4 +155,171 @@ public void remove()
         }
     );
   }
+
+  public static Filter convertToCNFFromQueryContext(Query query, Filter filter)
+  {
+    if (filter == null) {
+      return null;
+    }
+    boolean useCNF = (Boolean) query.getContextValue(CTX_KEY_USE_FILTER_CNF, false);
```

Comment:
I don't have a strong opinion on this, but I was leaning towards keeping it "false" by default since: 
- the user may not expect a filter transformation to occur
- the transformation doesn't always result in better performance (depending on the filter, it may be more efficient to evaluate the untransformed one)

So maybe it's better to have the user deliberately specify that the CNF transform should occur